app/assets/js/admin-upload-terms.js

```diff
@@ -0,0 +1,104 @@
+        rtp_terms_option_toggle();
+    });
+    jQuery('#bp-media-settings-boxes').on('submit', '#bp_media_settings_form, .rtmedia-settings-submit', function (e) {
+        var return_code = true;
+
+        if (return_code && general_enable_upload_terms.length > 0 && typeof general_enable_upload_terms != "undefined" ||
+            return_code && activity_enable_upload_terms.length > 0 && typeof activity_enable_upload_terms != "undefined" )
+        {
+            var error_msg = "";
+            if (general_enable_upload_terms.prop('checked') == true || activity_enable_upload_terms.prop('checked') == true)
+            {
+                jQuery( '.error_msg' ).remove();
+                jQuery( '.rtm-form-text' ).css('border-color', '#ddd');
+                if (!/^(http|https|ftp):\/\/[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/i.test(general_upload_terms_page_link.val())) {
+                    error_msg += 'Please enter valid URL.';
+                    return rtp_show_error_message ( general_upload_terms_page_link, error_msg );
+                }
+
+                /* Check "Terms of Service Message" Emply Or Not */
+                if ( general_upload_terms_message.val().trim() == '') {
+                     error_msg += 'Please enter terms message.';
+                    return rtp_show_error_message ( general_upload_terms_message, error_msg );
+                }
+
+                /* Check "Error Message" Emply Or Not */
+                if ( general_upload_terms_error_message.val().trim() == '') {
+                     error_msg += 'Please enter error message.';
+                    return rtp_show_error_message ( general_upload_terms_error_message, error_msg );
+                }
+            }
+        }
+        if(return_code && general_upload_terms_show_pricacy_message.length > 0 && typeof general_upload_terms_show_pricacy_message != "undefined") {
+            var error_msg = "";
+            if ( general_upload_terms_show_pricacy_message.prop('checked') ) {
+                jQuery( '.error_msg' ).remove();
+
+                /* Check "Terms of Service Message" Emply Or Not */
+                if ( general_upload_terms_privacy_message.val().trim() == '') {
+                     error_msg += 'Please enter privacy message.';
+                    return rtp_show_error_message ( general_upload_terms_privacy_message, error_msg );
+                }
+            }
+        }
+    });
+
+    /* Show Error Message If Incorrect Validation  */
+    function rtp_show_error_message ( selector, error_msg ) {
+        jQuery( selector ).focus();;
+        jQuery( selector ).css('border-color', 'red');
+        if ( jQuery( selector ).next().attr('class') != 'error_msg' ) {
+            jQuery( selector ).after('<span class="error_msg" style="display:block"> ' + error_msg + ' </span>');
+        }
```

Comment:
Take one variable having the element reference. I.e. `var elem = jQuery( selector );`, and then perform operations on it to prevent jQuery from multiple DOM queries. @malavasita